src/net_processing.cpp

```diff
@@ -1848,7 +1844,7 @@ static void ProcessHeadersMessage(CNode& pfrom, CConnman& connman, ChainstateMan
             UpdateBlockAvailability(pfrom.GetId(), headers.back().GetHash());
 
             if (nodestate->nUnconnectingHeaders % MAX_UNCONNECTING_HEADERS == 0) {
-                Misbehaving(pfrom.GetId(), 20);
+                Misbehaving(pfrom.GetId(), 20, strprintf("%d unconnecting headers", nodestate->nUnconnectingHeaders));
```

Comment:
2d4e694 the log message might be more clear for users if s/unconnecting/non-connecting/...yes, in the codebase it is called unconnecting...but nicely enough, anyone grepping for "non-connecting" will see this line as the only result and link it immediately to `nUnconnectingHeaders`

Reply:
(Maybe just me, but "unconnecting" strikes me as confusing naming; it suggests disconnection when we are really talking about non-connection.)

Reply:
(Maybe just me, but "unconnecting" strikes me as confusing naming; it suggests disconnection when we are really talking about invalid or non-connecting header messages, unless I am misreading the code in `ProcessHeadersMessage()` in net_processing.cpp and `CBlockIndex* LookupBlockIndex` in validation.cpp.)

Reply:
Agree that unconnecting sounds more like a verb than an adjective. I've changed this to non-connecting.